### ISSUE TITLE
multi-reduction scratch decision checks the wrong dimension

### ISSUE THREAD

**Issue by samanklesaria** [jax-ml/jax#39999]
### Description

In `jax/experimental/mosaic/gpu/dialect_lowering.py` in the `vector.multi_reduction` lowering, we see:
```python
  reduced_dim = src.layout.tiling.tile_dimension(op.reduction_dims[0])
  if any(reduced_dim[d] for d in src.layout.partitioned_warp_dims):
    # cross-warp reductions require scratch space.
    ...
  else:
    scratch = None
```
`Tiling.tile_dimension` indexes the tiling, but `op.reduction_dims[0]` is an array axis. The two are not the same: a `TiledLayout`'s tiling applies to an array's **trailing** dimensions, so when an array's rank exceeds the layout's base tile rank, array axis `a` corresponds to tiling axis `a - untiled_rank`, where `untiled_rank = len(shape) - len(layout.base_tile_shape)`.

Compare `FragmentedArray.reduce`, which decides whether the scratch is *needed*:

```python
    untiled_rank = len(self.shape) - len(layout.base_tile_shape)
    tiled_axes = tuple(a - untiled_rank for a in axis if a >= untiled_rank)
```

So for `(4, 128, 32)` reduced along axis 1, the caller inspects tiling axis 1 (the minor axis) while the reduction inspects tiling axis 0 (the rows). They disagree, the caller passes `scratch=None`, and the reduction raises `scratch must be provided when cross-warp reduction is required`.

Here's some code that shows the error in practice:

```python
from jax.experimental.mosaic.gpu import fragmented_array as fa

layout = fa.WGMMA_LAYOUT
warp_dims = layout.partitioned_warp_dims

for shape, axis in (((128, 32), 0), ((4, 128, 32), 1)):
  untiled_rank = len(shape) - len(layout.base_tile_shape)
  as_is = layout.tiling.tile_dimension(axis)               # the lowering rule
  rebased = layout.tiling.tile_dimension(axis - untiled_rank)  # reduce()
  print(f'{shape}, reducing axis {axis} (untiled_rank={untiled_rank}):')
  print(f'  lowering rule says scratch needed: {any(as_is[d] for d in warp_dims)}')
  print(f'  reduce() needs scratch:            {any(rebased[d] for d in warp_dims)}')
```

We get 
```
(128, 32), reducing axis 0 (untiled_rank=0):
  lowering rule says scratch needed: True
  reduce() needs scratch:            True
(4, 128, 32), reducing axis 1 (untiled_rank=1):
  lowering rule says scratch needed: False
  reduce() needs scratch:            True
```

### System info (python version, jaxlib version, accelerator, etc.)

jax:    0.11.1.dev20260814+ce6cfc5c2b
jaxlib: 0.11.0
numpy:  2.5.1
python: 3.13.5 (main, Jun 12 2025, 12:22:43) [Clang 20.1.4 ]
device info: cpu-1, 1 local devices"
process_count: 1
platform: uname_result(system='Darwin', release='23.5.0', version='Darwin Kernel Version 23.5.0: Wed May  1 20:14:59 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T8122', machine='arm64')
